tools/ci_connector_ops/ci_connector_ops/pipelines/commands/groups/metadata.py

```diff
@@ -37,9 +38,6 @@ def metadata(ctx: click.Context):
 def validate(ctx: click.Context, modified_only: bool) -> bool:
     if modified_only:
         metadata_to_validate = get_modified_metadata_files(ctx.obj["modified_files"])
```

Comment:
@alafanechere I think we have an error in this logic.

If a PR adds a new connector but no metadata file is found this will pass when it should fail.

We should add a new helper function
```py
def get_expected_metadata_file_paths(modified_files: Set[Union[str, Path]]) -> Set[Path]:
    # Any folder that matches airbyte-integrations/connectors/source-**, or airbyte-integrations/connectors/destination-**,
    # or airbyte-integrations/connectors/third-party/**/*/ should have a metadata file
    modified_connectors = get_modified_connectors(modified_files)
    modified_connector_file_paths = [connector.code_directory for connector in modified_connectors.keys()]
    expected_metadata_file_paths = [f"{connector_file_path}/{METADATA_FILE_NAME}" for connector_file_path in modified_connector_file_paths]
    return expected_metadata_file_paths;
```